### DISCUSSION TITLE
Harmonising jb2 desktop and web display inputs and config.json

### DISCUSSION THREAD

**Discussion by fubar2** [GMOD/jbrowse-components#4161] [General] [answered]
I did search but couldn't find any discussion about incompatibility between a working jbrowse2 config.json for a web server and that required for exactly the same directory structure by the desktop client?

For JB1, a downloaded tarball from our Galaxy client works perfectly well for the JB1 desktop.

Now, JB2 displays perfectly well in Galaxy, and when unpacked in an nginx static directory, the downloaded tarball displays as expected remotely. 
**However, the linux desktop client is completely baffled by the same config.json.** 
Wondering why this is the case and if there is any possibility of harmonising the two use cases? 
Inconvenient for users if they can't download a tarball for offsite/remote sharing with a desktop client - yes, they could serve it locally but the desktop client would be a far more convenient solution?

**Comment by cmdcolin** [comment, marked as answer]
this is a good idea but there are indeed a couple hurdles

we have an issue here tracking this

https://github.com/GMOD/jbrowse-components/issues/2190

we have also considered a way to even more tightly integrate desktop and web by making a system to open any web instance using a 'protocol handler' or something like this

https://github.com/GMOD/jbrowse-components/issues/2988

anyways, all this to say, this feature has been considered and would be nice, but does need some work to make it fully possible

**Reply by cmdcolin** [reply]
just in case you wanted to fix this for your use case, you could replace all {"uri":"somefile.ext"} with {"localPath":"somefile.ext"} in a new config.json for users to download

**Reply by fubar2** [reply]
> replace all {"uri":"somefile.ext"} with {"localPath":"somefile.ext"} 

@cmdcolin - thanks! That's handy to know if that's all that's needed for the desktop jb2 client. 

Currently testing to see if users can manage a tiny (5KB) byte-range python web server script that works well locally - it's copied into every jbrowse2 tarball root from the Galaxy tool, so the user can download and unzip it, change into the root and run:
`python servejb2.py` in that directory -it defaults to serving ./index.html on port 8080 or (use --port X to change) of localhost for browsing. Found the script [here](https://gist.github.com/glowinthedark/b99900abe935e4ab4857314d647a9068) - it will work for any jbrowse2 folder that can be served so can avoid the desktop client or the web admin server.
